### ISSUE TITLE
TypeError: Can't convert Hash into String.

### ISSUE THREAD

**Issue by ProgVal** [tootsuite/mastodon#6885]
Hi,

My Mastodon instance is (still) running v2.1.3. Several of my `ActivityPub::ProcessingWorker` jobs fail with this error:

```
Can't convert Hash into String.
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/addressable-2.5.2/lib/addressable/uri.rb:103:in `rescue in parse'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/addressable-2.5.2/lib/addressable/uri.rb:100:in `parse'
/home/mastodon-code/mastodon/app/services/activitypub/fetch_remote_status_service.rb:37:in `trustworthy_attribution?'
/home/mastodon-code/mastodon/app/services/activitypub/fetch_remote_status_service.rb:16:in `call'
/home/mastodon-code/mastodon/app/lib/activitypub/activity/announce.rb:31:in `fetch_remote_original_status'
/home/mastodon-code/mastodon/app/lib/activitypub/activity/announce.rb:6:in `perform'
/home/mastodon-code/mastodon/app/services/activitypub/process_collection_service.rb:43:in `process_item'
/home/mastodon-code/mastodon/app/services/activitypub/process_collection_service.rb:34:in `block in process_items'
/home/mastodon-code/mastodon/app/services/activitypub/process_collection_service.rb:34:in `reverse_each'
/home/mastodon-code/mastodon/app/services/activitypub/process_collection_service.rb:34:in `each'
/home/mastodon-code/mastodon/app/services/activitypub/process_collection_service.rb:34:in `map'
/home/mastodon-code/mastodon/app/services/activitypub/process_collection_service.rb:34:in `process_items'
/home/mastodon-code/mastodon/app/services/activitypub/process_collection_service.rb:21:in `call'
/home/mastodon-code/mastodon/app/workers/activitypub/processing_worker.rb:9:in `perform'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:188:in `execute_job'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:170:in `block (2 levels) in process'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/middleware/chain.rb:128:in `block in invoke'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-unique-jobs-5.0.10/lib/sidekiq_unique_jobs/server/middleware.rb:17:in `call'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/middleware/chain.rb:130:in `block in invoke'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/middleware/chain.rb:133:in `invoke'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:169:in `block in process'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:141:in `block (6 levels) in dispatch'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/job_retry.rb:97:in `local'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:140:in `block (5 levels) in dispatch'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/rails.rb:42:in `block in call'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/activesupport-5.1.4/lib/active_support/execution_wrapper.rb:85:in `wrap'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/activesupport-5.1.4/lib/active_support/reloader.rb:68:in `block in wrap'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/activesupport-5.1.4/lib/active_support/execution_wrapper.rb:85:in `wrap'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/activesupport-5.1.4/lib/active_support/reloader.rb:67:in `wrap'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/rails.rb:41:in `call'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:136:in `block (4 levels) in dispatch'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:204:in `stats'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:131:in `block (3 levels) in dispatch'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/job_logger.rb:7:in `call'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:130:in `block (2 levels) in dispatch'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/job_retry.rb:72:in `global'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:129:in `block in dispatch'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/logging.rb:44:in `with_context'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/logging.rb:38:in `with_job_hash_context'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:128:in `dispatch'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:168:in `process'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:85:in `process_one'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/processor.rb:73:in `run'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/util.rb:16:in `watchdog'
/home/mastodon-code/mastodon/vendor/bundle/ruby/2.3.0/gems/sidekiq-5.0.5/lib/sidekiq/util.rb:25:in `block in safe_thread'
```

Arguments of the job:

>` "args":[10940,"{\"@context\":[\"https://www.w3.org/ns/activitystreams\",\"https://w3id.org/security/v1\",{\"manuallyApprovesFollowers\":\"as:manuallyApprovesFollowers\",\"sensitive\":\"as:sensitive\",\"movedTo\":\"as:movedTo\",\"Hashtag\":\"as:Hashtag\",\"ostatus\":\"http://ostatus.org#\",\"atomUri\":\"ostatus:atomUri\",\"inReplyToAtomUri\":\"ostatus:inReplyToAtomUri\",\"conversation\":\"ostatus:conversation\",\"toot\":\"http://joinmastodon.org/ns#\",\"Emoji\":\"toot:Emoji\",\"focalPoint\":{\"@container\":\"@list\",\"@id\":\"toot:focalPoint\"},\"featured\":\"toot:featured\"}],\"id\":\"https://masto.svnet.fr/users/svnetlibre/statuses/99724151258668642/activity\",\"type\":\"Announce\",\"actor\":\"https://masto.svnet.fr/users/svnetlibre\",\"published\":\"2018-03-21T22:03:08Z\",\"to\":[\"https://www.w3.org/ns/activitystreams#Public\"],\"cc\":[\"https://tube.svnet.fr/accounts/svnet\",\"https://masto.svnet.fr/users/svnetlibre/followers\"],\"object\":\"https://tube.svnet.fr/videos/watch/61147603-a788-432a-bfc7-eb04d39e28b4\",\"atomUri\":\"https://masto.svnet.fr/users/svnetlibre/statuses/99724151258668642/activity\",\"signature\":{\"type\":\"RsaSignature2017\",\"creator\":\"https://masto.svnet.fr/users/svnetlibre#main-key\",\"created\":\"2018-03-21T22:03:08Z\",\"signatureValue\":\"<redacted>\"}}"]`

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by ThibG** [comment]
Weird, I can't understand why it would fail so noisily on 2.1.3.
In any case, the status would not have been accepted by Mastodon as it does not use `attributedTo`. This might be a bug in PeerTube.

**Comment by nightpool** [comment]
@ThibG Announce doesn't have attributedTo, only Create does.

**Comment by nightpool** [comment]
The problem seems to be that the referenced Video is attributed to *two* different actors:

```
"attributedTo": [
    {
      "type": "Person",
      "id": "https://tube.svnet.fr/accounts/svnet"
    },
    {
      "type": "Group",
      "id": "https://tube.svnet.fr/video-channels/d785b645-a961-43be-bc4f-519eb3279818"
    }
  ],
```
which we don't handle property.

**Comment by nightpool** [comment]
This issue was fixed in release 2.2.0, on January 28th
